File: kotlin/lesson 2/LinkedListStack.kt
class LinkedListStack<T> : Stack<T> {
    var head: Node<T>? = null
    
    override fun push(value: T) {
        head = Node(value, head)
    }

    override fun pop(): T {
        val result = head?.value ?: throw Exception("Stack is empty")
        head = head?.next
        return result
    }
}

File: csharp/lesson 2/LinkedListStack.cs
public class LinkedListStack<T> : Stack<T>
{
    private Node<T> head = null;

    public void Push(T value) 
    {
        head = new Node<T>(value, head);
    }

    public T Pop() 
    {
        T result = head.Value;
        head = head.Next;
        return result;
    }
}
